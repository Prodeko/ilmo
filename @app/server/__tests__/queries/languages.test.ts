import { deleteTestData, runGraphQLQuery, setup, teardown } from "../helpers"

beforeEach(deleteTestData)
beforeAll(setup)
afterAll(teardown)

test("languages query", async () => {
  await runGraphQLQuery(
    // GraphQL query goes here:
    `{
      languages {
        defaultLanguage
        supportedLanguages
      }
    }`,

    // GraphQL variables:
    {},

    // Additional props to add to `req` (e.g. `user: {session_id: '...'}`)
    {},

    // This function runs all your test assertions:
    async (json) => {
      expect(json.errors).toBeFalsy()
      expect(json.data).toBeTruthy()
      expect(json.data!.languages.defaultLanguage).toBe("fi")
      expect(json.data!.languages.supportedLanguages).toEqual(["fi", "en"])
    }
  )
})
