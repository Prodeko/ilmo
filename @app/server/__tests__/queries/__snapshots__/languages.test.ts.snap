// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languages query 1`] = `
Object {
  "data": Object {
    "defaultLanguage": Array [
      "FI",
    ],
    "supportedLanguages": Array [
      "FI",
      "EN",
      "SE",
    ],
  },
}
`;
